app/js/sumo_db.js

```diff
@@ -122,6 +141,22 @@
       return Promise.all([vote_promise, solved_promise]);
     },
 
+    set_category_for_question: function(category, question_id) {
+      var promise = SumoDB.get_question(question_id);
+      
+      return promise.then(function(question) {
+        var category_tags = question.tags.filter(function(tag) {
+          return tag.name.startsWith(CATEGORY_PREFIX);
+        });
+        category_tags = category_tags.map(function(elem) {
```

Comment:
Instead of repeating the variable, you can just chain the map call after the filter one.